modules/optimization/src/base/OptimizationApp.C

```diff
@@ -0,0 +1,74 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "OptimizationApp.h"
+#include "HeatConductionApp.h"
+#include "StochasticToolsApp.h"
+#include "Moose.h"
+#include "AppFactory.h"
+#include "MooseSyntax.h"
+#include "ExecFlagRegistry.h"
+
+#include "OptimizationAppTypes.h"
+
+InputParameters
+OptimizationApp::validParams()
+{
+  InputParameters params = MooseApp::validParams();
+
+  // Do not use legacy DirichletBC, that is, set DirichletBC default for preset = true
+  params.set<bool>("use_legacy_material_output") = false;
+
+  return params;
+}
+
+OptimizationApp::OptimizationApp(InputParameters parameters) : MooseApp(parameters)
+{
+  OptimizationApp::registerAll(_factory, _action_factory, _syntax);
+}
+
+OptimizationApp::~OptimizationApp() {}
+
+void
+OptimizationApp::registerAll(Factory & f, ActionFactory & af, Syntax & s)
+{
+  Registry::registerObjectsTo(f, {"OptimizationApp"});
+  Registry::registerActionsTo(af, {"OptimizationApp"});
+
+  auto & syntax = s;
```

Comment:
```suggestion
```

Reply:
It needs this to register the stochastic tools syntax.  Do I not need to register stochastic tools syntax since it is now a module?

Reply:
I think syntax is not used below? only s?

Reply:
it's used by a macro

Reply:
gross